runtime/src/snapshot_utils/snapshot_storage_rebuilder.rs

```diff
@@ -226,14 +225,25 @@ impl SnapshotStorageRebuilder {
     /// Spawn a single thread to process received append_vec_files
     fn spawn_receiver_thread(
         thread_pool: &ThreadPool,
-        exit_sender: Sender<()>,
+        exit_sender: Sender<Result<(), std::io::Error>>,
         rebuilder: Arc<SnapshotStorageRebuilder>,
     ) {
         thread_pool.spawn(move || {
             for path in rebuilder.file_receiver.iter() {
-                rebuilder.process_append_vec_file(path).unwrap();
+                match rebuilder.process_append_vec_file(path) {
+                    Ok(_) => {}
+                    Err(err) => {
+                        if let Err(err) = exit_sender.send(Err(err)) {
+                            error!("Failed to send error to exit_sender: {err:?}");
+                        }
+                        return;
```

Comment:
Agreed something is quite wrong if we cannot send.

I did this as what could be going wrong (https://github.com/solana-labs/solana/issues/29053#issuecomment-1342992328) might be the main-thread exitted for some reason and our file went out of scope, which potentially means the receiver also went out of scope. I think it's best that these worker threads are effectively unable to panic.

Though it might be overkill here